src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -264,39 +352,78 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return err
 	}
 
-	var ordinal int64
+	var ordinal int32
 	ordinal, err = utils.GetPodOrdinal(pod.Name, r.pandaCluster.Name)
 	if err != nil {
 		return fmt.Errorf("cluster %s: cannot convert pod name (%s) to ordinal: %w", r.pandaCluster.Name, pod.Name, err)
 	}
 
-	if patchResult.IsEmpty() {
-		if err = r.checkMaintenanceMode(ctx, int32(ordinal)); err != nil {
+	managedDecommission, err := r.IsManagedDecommission()
+	if err != nil {
+		log.Error(err, "not performing a managed decommission")
+	}
+
+	isRedpandaConfigUpdate := utils.IsStatusPodConditionTrue(pod.Status.Conditions, ClusterUpdatePodCondition)
+
+	// if the pod definition changes, patchResult is not empty
+	// if this isn't a managedDecommission
+	// if the pod isn't needing to be updated for redpanda
+	// take the pod out of maintenance mode
+	if patchResult.IsEmpty() && !managedDecommission && !isRedpandaConfigUpdate {
```

Comment:
I just realised that this part of the code will be never called. 

Events are as follow:
1. put condition on all pods
2. delete pod aka remove condition from the pod
3. `rollingUpdate` will never create list of the already delete pod from step 2.

If condition is not created in the Pod status, then PodEviction will never be called on that particular Pod.

Reply:
The `checkMaintenanceMode` should be moved somewhere else.

More info why it's important https://github.com/redpanda-data/redpanda/commit/421ce9c1

Reply:
I would suggest to move the `checkMaintenanceMode` in https://github.com/redpanda-data/redpanda/blob/c99ff31ad11fa1fd677309ebd0ac211edcae815b/src/go/k8s/pkg/resources/statefulset_update.go#L244-L249